### PR TITLE
Implement stack ADT

### PR DESCRIPTION
**Pull request by souvikhaldar**

### CODE REVIEW

File: src/stack_in_golang.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+package main
+import "fmt"
+type Stack interface{
```
Comment (rajkumarGosavi) [inline src/stack_in_golang.go]: comment needed to explain the interface

**Review by rajkumarGosavi — COMMENTED** [review]
Needs commenting so that learner can understand the use of the variables and methods implemented

File: src/stack_in_golang.go (hunk 1 above)
Comment (souvikhaldar) [reply src/stack_in_golang.go]: > comment needed to explain the interface

The interface is an explanation of the functionalities that any type which implements this holds. 
Make it simple, but no simpler.

Comment (sagar-jadhav) [reply src/stack_in_golang.go]: Add description of program as comments

Comment (souvikhaldar) [reply src/stack_in_golang.go]: Added comments.

File: src/stack_in_golang.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+package main
```
Comment (sagar-jadhav) [inline src/stack_in_golang.go]: Rename file to `stack.go`

**Review by sagar-jadhav — CHANGES_REQUESTED** [review]
Link `stack.go` in `README`

File: src/stack_in_golang.go (hunk 2 above)
Comment (souvikhaldar) [reply src/stack_in_golang.go]: Done

File: README.md [hunk 3]
```diff
@@ -32,6 +32,8 @@ To perform hands-on on **go-examples** it is prerequisite to have Go installed o
 * [Struct](https://github.com/sagar-jadhav/go-examples/blob/master/src/struct.go)
 * [Switch Statement](https://github.com/sagar-jadhav/go-examples/blob/master/src/switch.go)
 * [Unit Test](https://github.com/sagar-jadhav/go-examples/tree/master/src/test/power_test.go)
+* [Variadic Parameter](https://github.com/sagar-jadhav/go-examples/blob/master/src/variadic.go)
```
Comment (sagar-jadhav) [inline README.md:35]: Update link as `./variadic.html`

**Comment by sagar-jadhav** [comment]
@rajkumarGosavi & @aprabhat Please have a look at changes once.

File: README.md (hunk 3 above)
Comment (souvikhaldar) [reply README.md:35]: I do not understand your comment.
BTW, I've not made any changes regarding the above-mentioned line. @sagar-jadhav

Comment (sagar-jadhav) [reply README.md:35]: I agree suggested change is not related to your **change set**. 

I suggested this change to fix the broken link in README.

Comment (sagar-jadhav) [reply README.md:35]: Do you want me to create separate issue for this?

Comment (souvikhaldar) [reply README.md:35]: No, it was a small change so I've fixed it.

**Review by rajkumarGosavi — APPROVED** [review]

**Comment by souvikhaldar** [comment]
This PR had been reviewed, kindly merge.

**Review by sagar-jadhav — APPROVED** [review]

**Comment by sagar-jadhav** [comment]
> This PR had been reviewed, kindly merge.

Thank you @souvikhaldar for contribution 👍

**Comment by souvikhaldar** [comment]
A quick question, would this count towards hacktoberfest?

**Comment by sagar-jadhav** [comment]
> PRs can be made to any public repo on GitHub, not only the ones with issues labeled Hacktoberfest

**Reference:** [https://hacktoberfest.digitalocean.com/](https://hacktoberfest.digitalocean.com/)

Yes @souvikhaldar your contribution will be counted towards `hacktoberfest` once again thank you for participating in `Hacktoberfest 19`

**Pull request merged**